examples/fishnet_api/main.py

```diff
@@ -102,16 +118,58 @@ async def get_executions() -> List[Execution]:
     return await Execution.fetch_all()
 
 
+@app.get("/executions/{dataset_ID}")
+async def get_executions_by_dataset(dataset_ID: str) -> List[Execution]:
+    return await Execution.query(datasetID=dataset_ID)
+
+
 @app.get("/user/{address}/executions")
 async def get_user_executions(address: str) -> List[Execution]:
     return await Execution.query(owner=address)
 
 
+@app.get("/Permission")
+async def permission():
+    status = []
+    permission = await Permission.fetch_all()
+    for i in permission:
+        status.append(i.status)
+    if all(element == PermissionStatus.GRANTED for element in status):
+        return PermissionStatus.GRANTED
+    elif PermissionStatus.DENIED in status:
+        return PermissionStatus.DENIED
+    elif PermissionStatus.REQUESTED in status:
+        return PermissionStatus.REQUESTED
+
+
 @app.get("/user/{address}/results")
-async def get_user_executions(address: str) -> List[Result]:
+async def get_user_results(address: str) -> List[Result]:
     return await Result.query(owner=address)
 
 
+@app.get("/dataset/")
+async def get_requestor(requestor: str):
```

Comment:
It would be better to modify the `GET /datasets/` endpoint such that you can pass this requestor parameter and then calculate the permissions for every dataset based on this requestor query parameter.